### ISSUE TITLE
window key-callback should be mutable

### ISSUE THREAD

**Issue by fecundf** [YellowApple/p6-GLFW#3]
The test code fails with the error _"Cannot assign to a readonly variable or a value"_ until I comment out `$window.key-callback = &key-callback`

Perhaps key-callback needs to be marked as read-write?

**Comment by YellowApple** [comment]
Weird; it's already marked with `is rw`, so I'd expect that to work (and at one point it was indeed working).  Does `GLFW::set-window-key-callback($window, &key-callback);` work?

**Comment by fecundf** [comment]
`GLFW::set-window-key-callback` gives
   Could not find symbol '&set-window-key-callback' in 'GLFW'

raku --version tells me
Welcome to 𝐑𝐚𝐤𝐮𝐝𝐨™ v2020.10.
Implementing the 𝐑𝐚𝐤𝐮™ programming language v6.d.
Built on MoarVM version 2020.10.

The META6.json looks like the same as in the repo, GLFW 0.1.0, source-url git://github.com/YellowApple/p6-GLFW.git

**Comment by fecundf** [comment]
This call doesn't crash `GLFW::Window::set-key-callback($window, &key-callback);` - I still am figuring out which key is 256!

I see `method key-callback() is rw` in GLFW::Window source code, not sure why it complains.

**Comment by YellowApple** [comment]
Howdy!  Sorry this took 3 years to fix, but it's indeed fixed in v0.1.1.

Long story short: at some point between when I originally wrote this module and when you tried to use it, Perl6/Raku started to require `return-rw` for `Proxy` objects intended to be mutable.

Before:

```raku
method key-callback() is rw {
    return Proxy.new(
        FETCH => sub ($) {},  # FIXME: no GLFW equivalent
        STORE => sub ($, &callback) {
            set-key-callback(self, &callback);
        });
}
```

After:

```raku
method key-callback() is rw {
    return-rw Proxy.new(
        FETCH => sub ($) {},  # FIXME: no GLFW equivalent
        STORE => sub ($, &callback) {
            set-key-callback(self, &callback);
        });
}
```

Side note:

> I still am figuring out which key is 256!

That's the ESC key :)  Full list is in `lib/GLFW/Key.rakumod`.
